examples/mixed_signal/tests/test_regulator.py

```diff
@@ -0,0 +1,242 @@
+                       current=current)
+    
+    @cocotb.coroutine
+    def _get_single_sample(self, node):
+        """Measure a single voltage/current pair on *node*."""
+        toggle = next(self._bittogglestream)
+        self.dut.i_analog_probe.node_to_probe <= node
+        self.analog_probe.probe_voltage_toggle <= toggle
+        self.analog_probe.probe_current_toggle <= toggle
+        yield Timer(5, units='ps')  # NOTE: needs some time for some reason
+        dataset = self.probe_values()
+        self.dut._log.debug(f"trim value={self.dut.trim_val.value.signed_integer}: {self.analog_probe.node_to_probe}={dataset.voltage} V, {dataset.current} A")
+        return dataset
+        
+    @cocotb.coroutine
+    def get_sample_data(self, nodes, num=1, delay_ns=1):
+        """For all *nodes*, get *num* samples, spaced *delay_ns* apart.
+
+        Yields:
+            list: List (*num* samples long) of :any:`Dataset` for all *nodes*.
+        """
+        if not isinstance(nodes, list):  # single element? make it a list
+            _nodes = [nodes]
+        else:
+            _nodes = nodes
+        datasets = []
+        for idx in range(num):
+            for node in _nodes:
+                dataset = yield self._get_single_sample(node)
+                datasets.append(dataset)
+                if num > 1 and idx < num:
+                    yield Timer(delay_ns, units='ns')
+        return datasets
+
+    @cocotb.coroutine
+    def find_trim_val(self, probed_node, target_volt, trim_val_node, trim_val_repr='TWOS_COMPLEMENT'):
+        """Calculate best trimming value for *target_volt*.
+        Assumes a linear behaviour.
+
+        Args:
+            probed_node: The node to probe for the trimmed voltage.
+            target_volt (float): The target voltage at *probed_node*.
+            trim_val_node: The node to apply the trim value to.
+            trim_val_repr (str, optional): String indicating whether the *trim_val_node* has an
+                'UNSIGNED' or 'TWOS_COMPLEMENT' representation.
+
+        Yields:
+            float: The calculated best value for *trim_val_node*.
+        """
+        # find trim_val_min/trim_val_max based on bit length:
+        assert trim_val_repr in ('UNSIGNED', 'TWOS_COMPLEMENT')
+        trim_val_min = min_val(trim_val_node, trim_val_repr)
+        trim_val_max = max_val(trim_val_node, trim_val_repr)
+        # the actual trimming procedure:
+        trim_val_node <= trim_val_min
+        yield Timer(self.settling_time_ns, units='ns')
+        sample = yield self.get_sample_data(probed_node)
+        volt_min = sample[0].voltage
+        trim_val_node <= trim_val_max
+        yield Timer(self.settling_time_ns, units='ns')
+        sample = yield self.get_sample_data(probed_node)
+        volt_max = sample[0].voltage
+        if target_volt > volt_max:
+            self.dut._log.debug(f"target_volt={target_volt} > volt_max={volt_max}, returning minimum trim value {trim_val_max}")
```

Comment:
You're using 3.6-only syntax here. Given that, I'd argue you may as well use `async` / `await` for this example.

Reply:
Thanks for the hint, I actually meant to rewrite this to work without f-strings but forgot.